packages/salesforcedx-vscode-core/src/commands/baseDeployCommand.ts

```diff
@@ -59,11 +62,11 @@ export abstract class BaseDeployExecutor extends SfdxCommandletExecutor<
 
     execution.processExitSubject.subscribe(async exitCode => {
       let properties;
-      const registryAccess = new RegistryAccess();
       try {
-        const components = registryAccess.getComponentsFromPath(
-          execFilePathOrPaths
-        );
+        const ws = new WorkingSet();
+        const comps = ws.resolveSourceComponents(execFilePathOrPaths);
```

Comment:
```suggestion
        const ws = WorkingSet.fromSource(execFilePathOrPaths);
```

Shorthand for initializing WorkingSet with source